<a href="https://colab.research.google.com/github/KwasiArhin/Thinkful-assignments-/blob/master/daskarray.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import dask.array as da

## 1. Change the code above by setting `chunks=(250, 250)`. How long does it take to run?

In [5]:
%%timeit
x = da.random.random((10000, 10000), chunks=(250, 250))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1 loop, best of 3: 1.94 s per loop


It took around 1.94 seconds or 1940 milliseconds.

## 2. Now, set the parameter to `chunks=(500, 500)`. How long does it take to run? Does this one or the previous one run quickly? Why?

In [6]:
%%timeit
x = da.random.random((10000, 10000), chunks=(500, 500))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1 loop, best of 3: 1.03 s per loop


It took around 1030 milliseconds which is around half of what it takes in the previous task. This is because, when we divide a Dask array into smaller chunks Dask needs to manage a higher number of chunks. This requires more computation. If our dataset can fit into memory as is in our case, then setting the chunk size to a larger number speeds up the computation.